### PR TITLE
Remove some open in the implementation of the Stdlib

### PR DESCRIPTION
**Pull request by hhugo**
Using `open` at toplevel can make code harder to follow.

The PR removes all toplevel `open`s and use qualified names instead but 
for `open CamlinternalFormatBasics` and `open CamlinternalFormat`

- remove `open Obj` from `camlinternalOO.ml`. There were previously a mix of `Obj.obj` and `obj`.
- remove `open Lexing` from `parsing.ml` (introduce a `local open Lexing`)
- remove `open Printf` from `arg.ml`, `gc.ml`, `printexc.ml`
 
After this PR, there are two local open left

```
$ ag "[^_]open [A-Z]" stdlib/ -G ml$
stdlib/parsing.ml
129:  let open Lexing in

stdlib/format.ml
1264:
1265:open CamlinternalFormatBasics
1266:open CamlinternalFormat

stdlib/scanf.ml
15:
16:open CamlinternalFormatBasics
17:open CamlinternalFormat

stdlib/camlinternalFormat.ml
15:
16:open CamlinternalFormatBasics

stdlib/printf.ml
15:
16:open CamlinternalFormatBasics
17:open CamlinternalFormat

stdlib/obj.ml
79:    let open Nativeint in
```

### CODE REVIEW

File: stdlib/gc.ml [hunk 1]
```diff
@@ -67,31 +67,29 @@ external huge_fallback_count : unit -> int = "caml_gc_huge_fallback_count"
 external eventlog_pause : unit -> unit = "caml_eventlog_pause"
 external eventlog_resume : unit -> unit = "caml_eventlog_resume"
 
-open Printf
-
 let print_stat c =
   let st = stat () in
```
Comment (dra27) [inline stdlib/gc.ml:68]: This one is possibly nicer with an auxiliary similar to the one you temporarily introduced in the commit history:

```suggestion
  let st = stat () in
  let print fmt = Printf.fprintf c (fmt ^^ "\n") in
```

**Review by dra27 — COMMENTED** [review]
FWIW, I prefer fewer `open`s too!

File: stdlib/gc.ml (hunk 1 above)
Comment (hhugo) [reply stdlib/gc.ml:68]: I've applied your suggestion

**Comment by hhugo** [comment]
I've just rebased this PR and removed  `open Bigarry` from `Random`

**Comment by xavierleroy** [comment]
Quoting from the Bigarray documentation:
```
   Users of this module are encouraged to do [open Bigarray] in their
   source, then refer to array types and operations via short dot
   notation, e.g. [Array1.t] or [Array2.sub].
```
The Bigarray module was designed to be opened...

**Comment by hhugo** [comment]
Rebased again and dropped the commit related to bigarray based on @xavierleroy 's review.

**Review by xavierleroy — COMMENTED** [review]
For what is worth: I agree that the `open Obj` in `camlinternalOO.ml` is not great and explicit `Obj.` prefixing is better.  In all other cases I find the original code (with well-chosen `open` declarations) more legible than the modified code.  It's a matter of taste, so I'll shut up if there's a strong consensus in favor of the modified code.

**Comment by hhugo** [comment]
Closing. I've extracted the patch related to `open Obj` in #11520

**Pull request closed without merging**